libsolidity/formal/SymbolicTypes.cpp

```diff
@@ -544,17 +544,9 @@ void setSymbolicUnknownValue(smtutil::Expression _expr, frontend::TypePointer co
 smtutil::Expression symbolicUnknownConstraints(smtutil::Expression _expr, frontend::TypePointer const& _type)
 {
 	solAssert(_type, "");
-	if (isEnum(*_type))
+	if (isEnum(*_type) || isInteger(*_type) || isAddress(*_type) || isFixedBytes(*_type))
 	{
-		auto enumType = dynamic_cast<frontend::EnumType const*>(_type);
-		solAssert(enumType, "");
-		return _expr >= 0 && _expr < enumType->numberOfMembers();
-	}
-	else if (isInteger(*_type))
-	{
-		auto intType = dynamic_cast<frontend::IntegerType const*>(_type);
-		solAssert(intType, "");
-		return _expr >= minValue(*intType) && _expr <= maxValue(*intType);
+		return _expr >= minValue(_type) && _expr <= maxValue(_type);
```

Comment:
Do `minValue` and `maxValue` support all these types?

Reply:
Yes, they do. And it also seems to me they support them correctly :)

Reply:
Maybe `Enum` is not supported correctly. In `maxValue`:
```
if (auto const* enumType = dynamic_cast<EnumType const*>(_type))
		return enumType->numberOfMembers();
```
But to me it seems this should be `enumType->numberOfMembers() - 1`, no?

Reply:
True, that makes sense

Reply:
Fixed that and added a test that checks possible values of `Enum`.